samples/dotnet/openapi-skills/Program.cs

```diff
@@ -0,0 +1,227 @@
+            authCallback: authenticationProvider.AuthenticateRequestAsync);
+
+        // Use a planner to decide when to call the GitHub skill. Since we are not chaining operations, use
+        // the ActionPlanner which is a simplified planner that will always return a 0 or 1 step plan.
+        ActionPlanner planner = new(kernel, logger: logger);
+
+        // Chat loop
+        IChatCompletion chatGPT = kernel.GetService<IChatCompletion>();
+        OpenAIChatHistory chatHistory = (OpenAIChatHistory)chatGPT.CreateNewChat("You are a helpful, friendly, intelligent assistant that is good at conversation.");
+        while (true)
+        {
+            Console.WriteLine("----------------");
+            Console.Write("Input: ");
+            string? input = Console.ReadLine();
+
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                continue;
+            }
+
+            // Add GitHub's response, if any, to the chat history.
+            int planResultTokenAllowance = (int)(aiOptions.TokenLimit * 0.25); // Allow up to 25% of our token limit to be from GitHub.
+            string planResult = await PlanGitHubSkill(gitHubOptions, planner, chatHistory, input, planResultTokenAllowance);
+            if (!string.IsNullOrWhiteSpace(planResult))
+            {
+                chatHistory.AddUserMessage(planResult);
+            }
+
+            // Add the user's input to the chat history.
+            chatHistory.AddUserMessage(input);
+
+            // Remove earlier messages until we are back within our token limit.
+            // (Note this sample does not implement long-term memory)
+            int tokenCount = GPT3Tokenizer.Encode(JsonSerializer.Serialize(chatHistory)).Count;
+            while (tokenCount > aiOptions.TokenLimit)
+            {
+                chatHistory.Messages.RemoveAt(0);
+                tokenCount = GPT3Tokenizer.Encode(JsonSerializer.Serialize(chatHistory)).Count;
+            }
+            Console.WriteLine($"(tokens: {tokenCount})");
+
+            // Send the chat history to the AI for a response.
+            string reply = await chatGPT.GenerateMessageAsync(chatHistory);
+            chatHistory.AddAssistantMessage(reply);
+
+            Console.WriteLine("----------------");
+            Console.WriteLine($"Assistant: {reply}");
+        }
+    }
+
+    /// <summary>
+    /// Run the planner to decide whether to run the GitHub skill function and add the result to the chat history.
+    /// </summary>
+    private static async Task<string> PlanGitHubSkill(GitHubSkillOptions gitHubOptions, ActionPlanner planner, OpenAIChatHistory chatHistory, string input, int tokenAllowance)
+    {
+        // Ask the planner to create a plan based off the user's input. If the plan elicits no steps, continue normally.
+        Plan plan = await planner.CreatePlanAsync(input);
+        if (plan.Steps.Count > 0)
```

Comment:
Perhaps `plan.HasNextStep` instead?

Reply:
Really, you shouldn't even need to check. It will be a no-op if there is nothing to run.